### PR TITLE
Add lombok dependency,edit entities and edit used dao in controllers

### PR DESCRIPTION
**Pull request by luismiguel0529**
## Summary
- Add dependency lombok
- Edit entities with lombok
## Trello Card
https://trello.com/c/ZTFfaSuM/44-lombok-para-evitar-boiler-plate-code

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,17 +19,24 @@ dependencies {
...
 
+    testCompileOnly('org.projectlombok:lombok:1.18.16')
+    testAnnotationProcessor('org.projectlombok:lombok:1.18.16')
     testCompile('junit:junit:4.12')
     testCompile('com.github.tomakehurst:wiremock:2.27.2')
     testImplementation('com.h2database:h2:1.4.200')
     testImplementation('org.springframework.security:spring-security-test')
     testImplementation('org.springframework.boot:spring-boot-starter-test') {
-        exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
+        exclude group: 'org.junit.vintage', module: 'jun' +
```
Comment (facucompe) [inline build.gradle]: Leave it like before. I don't like this haha

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -25,6 +29,10 @@
  * @author luismiguelrodriguez
  */
 @Entity
+@Getter
```
Comment (facucompe) [inline src/main/java/wolox/training/models/Book.java]: Why not setters? use `@Data`

File: src/main/java/wolox/training/models/dto/BookDTO.java [hunk 3]
```diff
@@ -3,13 +3,22 @@
...
-import java.io.Serializable;
 import java.util.HashMap;
 import java.util.List;
 
-public class BookDTO implements Serializable {
+@Getter
+@Setter
+@NoArgsConstructor
+@ToString
+@EqualsAndHashCode
```
Comment (facucompe) [inline src/main/java/wolox/training/models/dto/BookDTO.java]: `@Data` does all this

**Review by facucompe — CHANGES_REQUESTED** [review]
Replace instantiation with builders

File: build.gradle (hunk 1 above)
Comment (luismiguel0529) [reply build.gradle]: And here what happened, I don't understand how it happened

File: src/main/java/wolox/training/models/Book.java (hunk 2 above)
Comment (luismiguel0529) [reply src/main/java/wolox/training/models/Book.java:32]: The setter methods are defined below in the entity, they use the pre conditions, please confirm me, if I should add them anyway

File: src/main/java/wolox/training/models/dto/BookDTO.java (hunk 3 above)
Comment (luismiguel0529) [reply src/main/java/wolox/training/models/dto/BookDTO.java]: corrected

File: src/main/java/wolox/training/models/Book.java (hunk 2 above)
Comment (luismiguel0529) [reply src/main/java/wolox/training/models/Book.java]: corrected

File: build.gradle (hunk 1 above)
Comment (luismiguel0529) [reply build.gradle]: corrected

File: src/main/java/wolox/training/models/Book.java [hunk 4]
```diff
@@ -80,107 +82,51 @@
...
-        this.year = year;
-        this.pages = pages;
-        this.isbn = isbn;
-    }
-
-    public Long getId() {
-        return id;
-    }
-
     public void setId(Long id) {
```
Comment (facucompe) [inline src/main/java/wolox/training/models/Book.java]: Remove this setters

Comment (facucompe) [reply src/main/java/wolox/training/models/Book.java]: Oh, i forgot preconditions.

**Review by facucompe — APPROVED** [review]

**Review by joredudiaz — APPROVED** [review]
👍

**Pull request merged**